### ISSUE TITLE
invalid json error for trace_filter requests at future blocks

### ISSUE THREAD

**Issue by oceja** [ledgerwatch/erigon#4978]
#### System information

Erigon version: `./erigon --version`

v2022.08.01

OS & Version: Windows/Linux/OSX
Linux

Commit hash : 

TLDR: When making the a trace_filter request for a future block, erigon returns invalid json

#### Expected behaviour

Valid json:
`{"jsonrpc":"2.0","id":0,"result":{"error":{"code":-32000,"message":"could not find block 0000000000000000000000000000000000000000000000000000000000000000 25310900"}}`

#### Actual behaviour

Invalid json:
`{"jsonrpc":"2.0","id":0,"result":[null,"error":{"code":-32000,"message":"could not find block 0000000000000000000000000000000000000000000000000000000000000000 25310900"}}`

Notice the `[null,`

#### Steps to reproduce the behaviour

Example curl request:

`curl -s --data '{"jsonrpc": "2.0", "method": "trace_filter", "params": [{"fromBlock": "0x18236B4", "toBlock": "0x18236B4"}], "id": 0}' -H "Content-Type: application/json" -X POST http://localhost:8545 | jq`

#### Backtrace

````
[backtrace]
````

**Comment by oceja** [comment]
Actually this might have been fixed by @dafky2000 https://github.com/ledgerwatch/erigon/pull/4894/files

**Comment by dafky2000** [comment]
@oceja, it should be! Let me know if not, I'll be happy to revisit.

**Comment by revittm** [comment]
Confirmed as fixed on latest devel, will close :)
